tests/phpunit/api/v4/Action/SqlExpressionTest.php

```diff
@@ -95,4 +96,31 @@ public function testSelectAlias() {
       ->execute();
   }
 
+  public function testSelectEquations() {
+    $contact = Contact::create(FALSE)->addValue('first_name', 'bob')
+      ->addChain('email', Email::create()->setValues(['email' => 'hello@example.com', 'contact_id' => '$id']))
+      ->execute()->first();
+    $result = Email::get(FALSE)
+      ->setSelect([
+        'IF((contact_id.first_name = "bob"), "Yes", "No") AS is_bob',
+        'IF((contact_id.first_name != "fred"), "No", "Yes") AS is_fred',
+        '(5 * 11)',
+        '(5 > 11) AS five_greater_eleven',
+        '(5 <= 11) AS five_less_eleven',
+        '(1 BETWEEN 0 AND contact_id) AS is_between',
+        '(illegal * stuff) AS illegal_stuff',
+      ])
+      ->addWhere('contact_id', '=', $contact['id'])
+      ->setLimit(1)
+      ->execute()
+      ->first();
+    $this->assertEquals('Yes', $result['is_bob']);
+    $this->assertEquals('No', $result['is_fred']);
+    $this->assertEquals('55', $result['5_11']);
+    $this->assertFalse($result['five_greater_eleven']);
+    $this->assertTrue($result['five_less_eleven']);
+    $this->assertTrue($result['is_between']);
+    $this->assertArrayNotHasKey('illegal_stuff', $result);
```

Comment:
I think it's OK to omit invalid items. Would it be hard to (or does it already) log a warning about why something gets dropped?

Reply:
Coleman mentioned in chat that there is a way to see the debug log and get some kind of hint that the field has failed.

So merge on pass.